OrbitGl/LiveFunctionsDataView.cpp

```diff
@@ -295,21 +293,23 @@ void LiveFunctionsDataView::DoFilter() {
   for (size_t i = 0; i < indices_.size(); ++i) {
     FunctionInfo* func = GetFunction(i);
     Capture::GVisibleFunctionsMap[FunctionUtils::GetAbsoluteAddress(*func)] =
-        func;
+        *func;
   }
 
   GOrbitApp->NeedsRedraw();
 }
 
 void LiveFunctionsDataView::OnDataChanged() {
   functions_.clear();
-  const std::vector<std::shared_ptr<orbit_client_protos::FunctionInfo>>&
+  const absl::flat_hash_map<uint64_t, orbit_client_protos::FunctionInfo>&
       selected_functions = Capture::capture_data_.selected_functions();
   size_t functions_count = selected_functions.size();
   indices_.resize(functions_count);
-  for (size_t i = 0; i < functions_count; ++i) {
+  size_t i = 0;
+  for (const auto& pair : selected_functions) {
+    functions_.push_back(pair.second);
     indices_[i] = i;
-    functions_.push_back(selected_functions[i]);
+    i++;
```

Comment:
More for fun than anything, it's finally time to link this new addition to the styleguide: https://google.github.io/styleguide/cppguide.html#Preincrement_and_Predecrement 😄

Reply:
Well I kind off disagree with the style guide, but well fine to change this :D